bundles/org.openhab.binding.venstarthermostat/src/main/java/org/openhab/binding/venstarthermostat/handler/VenstarThermostatHandler.java

```diff
@@ -0,0 +1,502 @@
+                    .withArguments(CONFIG_USERNAME).build());
+        }
+
+        if (StringUtils.isBlank(config.password)) {
+            log.warn("password is empty");
+            status.add(ConfigStatusMessage.Builder.error(CONFIG_PASSWORD).withMessageKeySuffix(EMPTY_INVALID)
+                    .withArguments(CONFIG_PASSWORD).build());
+        }
+
+        if (config.refresh == null || config.refresh < 10) {
+            log.warn("refresh is too small: {}", config.refresh);
+
+            status.add(ConfigStatusMessage.Builder.error(CONFIG_REFRESH).withMessageKeySuffix(REFRESH_INVALID)
+                    .withArguments(CONFIG_REFRESH).build());
+        }
+        return status;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+
+        if (getThing().getStatus() != ThingStatus.ONLINE) {
+            log.debug("Controller is NOT ONLINE and is not responding to commands");
+            return;
+        }
+
+        stopUpdateTasks();
+        if (command instanceof RefreshType) {
+            log.debug("Refresh command requested for " + channelUID);
+            stateMap.clear();
+            startUpdatesTask(0);
+        } else {
+            stateMap.remove(channelUID.getAsString());
+            if (channelUID.getId().equals(CHANNEL_HEATING_SETPOINT)) {
+                QuantityType<Temperature> quantity = commandToQuantityType(command, unitSystem);
+                int value = quantityToRoundedTemperature(quantity, unitSystem).intValue();
+
+                log.debug("Setting heating setpoint to {}", value);
+                setHeatingSetpoint(value);
+            } else if (channelUID.getId().equals(CHANNEL_COOLING_SETPOINT)) {
+                QuantityType<Temperature> quantity = commandToQuantityType(command, unitSystem);
+                int value = quantityToRoundedTemperature(quantity, unitSystem).intValue();
+
+                log.debug("Setting cooling setpoint to {}", value);
+                setCoolingSetpoint(value);
+            } else if (channelUID.getId().equals(CHANNEL_SYSTEM_MODE)) {
+                VenstarSystemMode value;
+                if(command instanceof StringType) {
+                    value = VenstarSystemMode.valueOf(((StringType)command).toString().toUpperCase());
+
+                } else {
+                    value = VenstarSystemMode.fromInt(((DecimalType)command).intValue());
+                }
+                log.debug("Setting system mode to  {}", value);
+                setSystemMode(value);
+                updateIfChanged(CHANNEL_SYSTEM_MODE_RAW, new StringType("" + value));
+
+            }
+            startUpdatesTask(UPDATE_AFTER_COMMAND_SECONDS);
+        }
+
+    }
+
+    @Override
+    public void dispose() {
+        stopUpdateTasks();
+        if (httpClient.isStarted()) {
+            try {
+                httpClient.stop();
+            } catch (Exception e) {
+                log.error("Could not stop HttpClient", e);
```

Comment:
This error doesn't seem to be a big problem as it happens when disposing the handler. So it can be log to debug.
```suggestion
                log.debug("Could not stop HttpClient", e);
```